src/raftstore/coprocessor/split_check/keys.rs

```diff
@@ -0,0 +1,267 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use raftstore::store::{util, Msg};
+use rocksdb::DB;
+use util::transport::{RetryableSendCh, Sender};
+
+use super::super::metrics::*;
+use super::super::{Coprocessor, KeyEntry, ObserverContext, SplitCheckObserver, SplitChecker};
+use super::Host;
+
+pub struct Checker {
+    max_keys: u64,
+    split_keys: u64,
+    current_keys: u64,
+    split_key: Option<Vec<u8>>,
+}
+
+impl Checker {
+    pub fn new(max_keys: u64, split_keys: u64) -> Checker {
+        Checker {
+            max_keys,
+            split_keys,
+            current_keys: 0,
+            split_key: None,
+        }
+    }
+}
+
+impl SplitChecker for Checker {
+    fn on_kv(&mut self, _: &mut ObserverContext, key: &KeyEntry) -> bool {
+        if key.is_commit_version() {
+            self.current_keys += 1;
+        }
+        if self.current_keys > self.split_keys && self.split_key.is_none() {
+            self.split_key = Some(key.key().to_vec());
+        }
+        self.current_keys > self.max_keys
+    }
+
+    fn split_key(&mut self) -> Option<Vec<u8>> {
+        if self.current_keys > self.max_keys {
```

Comment:
I mean, why not just return `self.split_key.take()`?
If `self.current_keys <= self.max_keys`, `self.split_key.take()` will be `None` anyway.

Reply:
yes, you are right. we check `self.current_keys > self.max_keys ` just in order to  keep consistent with `SizeCheckObserver`

Reply:
Oh, my mistake, `split_keys` and `max_keys` are different.